src/robotFactory.js

```diff
@@ -1,10 +1,90 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = {};
 
-class FlyingRobot {}
+    if ('x' in coords) {
+      this.coords.x = coords.x;
+    } else {
+      this.coords.x = 0;
+    }
```

Comment:
```suggestion
    this.coords.x = coords.x || 0
```
Please fix all cases